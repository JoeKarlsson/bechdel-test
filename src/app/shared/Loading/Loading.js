import React from 'react';
import styles from './Loading.scss';

const Loading = () => (
	<div>
		<span className={styles.Loader}>
			<div className={styles.Loader_indicator} >
				<h1>
					<span>Destroying the Patriarchy</span>
					<span className={styles.Loader_ellipsis} >
						<span className={styles.Loader_ellipsisDot}>.</span>
						<span className={styles.Loader_ellipsisDot}>.</span>
						<span className={styles.Loader_ellipsisDot}>.</span>
					</span>
				</h1>
			</div>
		</span>
	</div>
);

export default Loading;
