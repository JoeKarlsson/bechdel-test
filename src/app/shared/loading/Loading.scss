.Loader_indicator.fade-enter,
.Loader_content.fade-enter {
	opacity: 0;
}

.Loader_indicator.fade-enter-active,
.Loader_content.fade-enter-active {
	opacity: 1;
}

.Loader_indicator.fade-leave,
.Loader_content.fade-leave {
	opacity: 1;
}

.Loader_indicator.fade-leave-active,
.Loader_content.fade-leave-active {
	opacity: 0;
}

.Loader_content,
.Loader_indicator {
	transition: opacity 1s;
}

.Loader_ellipsisDot {
	padding: 0 3px 0 1px;
	opacity: 0;
	animation-duration: 3001ms;
	animation-iteration-count: infinite;
	animation-timing-function: linear;
}

@keyframes Loader_ellipsisDot1 {

	0% {
		opacity: 0;
	}

	25% {
		opacity: 0;
	}

	35% {
		opacity: 1;
	}

	100% {
		opacity: 1;
	}
}

@keyframes Loader_ellipsisDot2 {

	0% {
		opacity: 0;
	}

	50% {
		opacity: 0;
	}

	60% {
		opacity: 1;
	}

	100% {
		opacity: 1;
	}
}

@keyframes Loader_ellipsisDot3 {

	0% {
		opacity: 0;
	}

	75% {
		opacity: 0;
	}

	85% {
		opacity: 1;
	}

	100% {
		opacity: 1;
	}
}

.Loader_ellipsisDot:nth-child(1) {
	animation-name: Loader_ellipsisDot1;
}

.Loader_ellipsisDot:nth-child(2) {
	animation-name: Loader_ellipsisDot2;
}

.Loader_ellipsisDot:nth-child(3) {
	animation-name: Loader_ellipsisDot3;
}
