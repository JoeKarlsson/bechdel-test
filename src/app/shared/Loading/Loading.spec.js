import React from 'react';
import {
	shallow,
	configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Loading from './Loading';

configure({ adapter: new Adapter() });

describe('Loading', () => {
	let wrapper;

	global.requestAnimationFrame = (callback) => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<Loading />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<Loading />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
