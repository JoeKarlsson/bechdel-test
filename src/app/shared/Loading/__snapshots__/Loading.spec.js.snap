// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading rendering initial state is rendered 1`] = `
<div>
  <span
    className="Loader"
  >
    <div
      className="Loader_indicator"
    >
      <h1>
        <span>
          Destroying the Patriarchy
        </span>
        <span
          className="Loader_ellipsis"
        >
          <span
            className="Loader_ellipsisDot"
          >
            .
          </span>
          <span
            className="Loader_ellipsisDot"
          >
            .
          </span>
          <span
            className="Loader_ellipsisDot"
          >
            .
          </span>
        </span>
      </h1>
    </div>
  </span>
</div>
`;
